src/main/java/quentin/core/ChainContainer.java

```diff
@@ -22,11 +22,17 @@
     }
 
     protected void updateChain(Intersection newIntersection) {
-        Graph<Intersection, DefaultEdge> chainsOfColor = chains.get(newIntersection.getStone());
-        chainsOfColor.addVertex(newIntersection);
-        chainsOfColor.vertexSet().stream()
-                .filter(newIntersection::isOrthogonalTo)
-                .forEach(orthogonalIntersection -> chainsOfColor.addEdge(orthogonalIntersection, newIntersection));
+        Optional<Graph<Intersection, DefaultEdge>> chainsOfColor = Optional.ofNullable(chains.get(newIntersection.getStone()));
+        chainsOfColor.ifPresent(chains -> {
```

Comment:
`chains` variable into the stream has the name of the `chains` private final local variable of the ChainContainer class.
I suggest to rename the stream variable (chainsSet ?).